client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,469 @@
+use {
+    crate::{
+        connection_cache::ConnectionCache,
+        nonblocking::{rpc_client::RpcClient, tpu_client::TpuClient},
+        rpc_client::RpcClient as BlockingRpcClient,
+    },
+    bincode::serialize,
+    dashmap::DashMap,
+    solana_quic_client::{QuicConfig, QuicConnectionManager, QuicPool},
+    solana_rpc_client::spinner,
+    solana_rpc_client_api::request::MAX_GET_SIGNATURE_STATUSES_QUERY_ITEMS,
+    solana_sdk::{
+        hash::Hash,
+        message::Message,
+        signature::Signature,
+        signers::Signers,
+        transaction::{Transaction, TransactionError},
+    },
+    solana_tpu_client::{
+        nonblocking::tpu_client::set_message_for_confirmed_transactions,
+        tpu_client::{Result, TpuClientConfig, TpuSenderError},
+    },
+    std::{
+        sync::{
+            atomic::{AtomicU32, AtomicU64, Ordering},
+            Arc,
+        },
+        time::Duration,
+    },
+    tokio::{sync::RwLock, task::JoinHandle, time::Instant},
+};
+
+const BLOCKHASH_REFRESH_RATE: Duration = Duration::from_secs(10);
+const TPU_RESEND_REFRESH_RATE: Duration = Duration::from_secs(2);
+
+#[derive(Clone, Debug)]
+struct TransactionData {
+    last_valid_blockheight: u64,
+    message: Message,
+    index: usize,
+    serialized_transaction: Vec<u8>,
+}
+
+#[derive(Clone, Debug, Copy)]
+struct BlockHashData {
+    pub blockhash: Hash,
+    pub last_valid_blockheight: u64,
+}
+
+pub fn send_and_confirm_transactions_in_parallel_blocking<T: Signers + ?Sized>(
+    rpc_client: Arc<BlockingRpcClient>,
+    websocket_url: String,
+    messages: &[Message],
+    signers: &T,
+    resign_txs_count: Option<usize>,
+    with_spinner: bool,
+    send_over_tpu: bool,
+) -> Result<Vec<Option<TransactionError>>> {
+    let fut = send_and_confirm_transactions_in_parallel(
+        rpc_client.get_inner_client().clone(),
+        websocket_url,
+        messages,
+        signers,
+        resign_txs_count,
+        with_spinner,
+        send_over_tpu,
+    );
+    tokio::task::block_in_place(|| rpc_client.runtime().block_on(fut))
+}
+
+fn create_blockhash_data_updating_task(
+    rpc_client: Arc<RpcClient>,
+    blockhash_data_rw: Arc<RwLock<BlockHashData>>,
+    current_blockheight: Arc<AtomicU64>,
+) -> JoinHandle<()> {
+    tokio::spawn(async move {
+        loop {
+            if let Ok(blockhash_and_valid_height) = rpc_client
```

Comment:
I wonder if you can unpack here as well like `if let Ok((blockhash, last_valid_blockheight))